custom_components/waste_collection_schedule/waste_collection_schedule/source/preston_gov_uk.py

```diff
@@ -0,0 +1,134 @@
+from datetime import datetime,date
+from typing import Optional
+from requests import Session
+from bs4 import BeautifulSoup
+from waste_collection_schedule import Collection  # type: ignore[attr-defined]
+
+TITLE = "Preston City Council"
+DESCRIPTION = "Source for preston.gov.uk services for Preston City Council, UK."
+URL = "https://preston.gov.uk"
+SRV_URL = "https://selfservice.preston.gov.uk/service/Forms/FindMyNearest.aspx?Service=bins"
+
+TEST_CASES = {
+    "Test_001": {"street": "town hall, lancaster road"},
+    "Test_002": {"street": "", "uprn": "10002220003"},
+}
+
+ICON_MAP = {
+    "Commercial -  general waste:": "mdi:trash-can",
+    "Commercial -  cardboard and paper:": "mdi:note-text",
+    "Commercial - plastic cans and glass:": "mdi:recycle",
+    "Black/grey bin (general waste):": "mdi:trash-can",
+    "Yellow lidded recycling container (glass/cans/plastics):": "mdi:recycle",
+    "Red lidded recycling container (paper/card):": "mdi:note-text",
+    "Brown bin (garden waste):": "mdi:leaf",
+}
+
+HEADER_MAP = {
+    "Commercial -  general waste:": "General Waste (Commercial)",
+    "Commercial -  cardboard and paper:": "Cardboard (Commercial)",
+    "Commercial - plastic cans and glass:": "Plastic (Commercial)",
+    "Black/grey bin (general waste):": "General Waste (Black/Grey bin)",
+    "Yellow lidded recycling container (glass/cans/plastics):": "Glass/Cans/Plastics (Yellow bin)",
+    "Red lidded recycling container (paper/card):": "Cardboard/Paper (Red bin)",
+    "Brown bin (garden waste):": "Garden/Green Waste (Brown bin)",
+}
+
+PARAMS = {
+    "__VIEWSTATE": "",
+    "__VIEWSTATEGENERATOR": "",
+    "__EVENTVALIDATION": "",
+    "__EVENTTARGET": "ctl00$MainContent$btnSearch",
+    "__EVENTARGUMENT": "",
+    "__LASTFOCUS": "",
+    "__SCROLLPOSITIONX": "0",
+    "__SCROLLPOSITIONY": "0",
+
+    "ctl00$MainContent$hdnService": "bins",
+    "ctl00$MainContent$hdnActivityId": "",
+    "ctl00$MainContent$hdnPropertyEntityAddress": "",
+    "ctl00$MainContent$hdnPropertyEntityValue": "",
+    "ctl00$MainContent$txtAddress": "",
+    "ctl00$MainContent$hdnUPRN": "",
+}
+
+class Source:
+    def __init__(self, street = "", uprn = ""):
+        self._street = street
+        self._uprn = str(uprn)
+        self._session = Session()
```

Comment:
I don't think keeping the session for days is a good idea. You're doing every request anyway, I think it's a better idea to create a new Session object every time fetch is called

Reply:
Thanks. Committed your suggestion. move Session() from __init__() to _get()